### DISCUSSION TITLE
Embedded mode

### DISCUSSION THREAD

**Discussion by rossipedia** [Code-Forge-Net/Remix-Dev-Tools#2] [Ideas]
Prior art: React Query DevTools [Embedded Mode](https://tanstack.com/query/v4/docs/react/devtools#embedded-mode)

The floating button can be useful as a "get set up and get going", but some apps probably want a more sophisticated way of showing / hiding their dev tools that doesn't involve giving up screen real-estate.

Adding something like a `RemixDevToolsPanel` that we can put wherever would be super awesome.

Fantastic work on this, can't wait to really dig in :+1:

**Comment by AlemTuzlak** [comment]
So basically expose the insides to be accessible for embedding? Hmm that vould be possible, I wonder if there is a way to open it up in a separate chrome window while still keeping the original site context that would also be cool 🤔

**Comment by AlemTuzlak** [comment]
@rossipedia so I've been thinking about this, what would make more sense, to have a panel with all the tabs, or to have each tab exposed so you can inject whatever you want wherever you want? eg I expose the pageTab component and you use it wherever you want

**Reply by rossipedia** [reply]
My specific use case would be the entire panel, but without the show/hide floating button, or resizing support. I don't necessarily mind having the individual bits and pieces, but I don't know if that's going to be as useful as the whole thing being embeddable into another surface.
